app/src/app/[lng]/[inventory]/page.tsx

```diff
@@ -76,31 +76,35 @@ export default function Home({ params: { lng } }: { params: { lng: string } }) {
   const { t } = useTranslation(lng, "dashboard");
   const toast = useToast();
   const router = useRouter();
+  const { inventory: cityParam } = useParams();
+  const inventoryId = cityParam as string;
 
   // query API data
   // TODO maybe rework this logic into one RTK query:
   // https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#performing-multiple-requests-with-a-single-query
-  let defaultInventoryId: string | null = null;
+
   const { data: userInfo, isLoading: isUserInfoLoading } =
     api.useGetUserInfoQuery();
+
+  let defaultInventoryId: string | null = null;
   if (!isUserInfoLoading && userInfo) {
     defaultInventoryId = userInfo.defaultInventoryId;
 
     // TODO also add this to login logic or after email verification to prevent extra redirect?
     // if the user doesn't have a default inventory, redirect to onboarding page
-    if (!defaultInventoryId) {
+    if (!inventoryId) {
       // fixes warning "Cannot update a component (`Router`) while rendering a different component (`Home`)"
-      setTimeout(() => router.push("/onboarding"), 0);
+      setTimeout(() => router.push(`/${defaultInventoryId}`), 0);
```

Comment:
Update the comment in case you intend to always redirect here. Also the user might not have a `defaultInventoryId` set at this stage, so check if it is null and redirect to onboarding in that case (like before).